packages/plugin-ext/src/main/browser/file-system-main.ts

```diff
@@ -54,6 +57,21 @@ export class FileSystemMainImpl implements FileSystemMain, Disposable {
         }
     }
 
+    async $readFile(uriComponents: UriComponents): Promise<string> {
+        const uri = Uri.revive(uriComponents);
+        const resource = this.resourceResolver.resolve(new URI(uri));
+        return resource.readContents();
+    }
+
+    async $writeFile(uriComponents: UriComponents, content: string): Promise<void> {
+        const uri = Uri.revive(uriComponents);
+        const resource = this.resourceResolver.resolve(new URI(uri));
+        if (!resource.saveContents) {
+            throw new FileSystemError(`'No write operation available on the resource for URI ${uriComponents}`);
```

Comment:
maybe return a error code and translate it to error in the plugin code to avoid pulling plugin code in bundle.js